crates/common_utils/src/pii.rs

```diff
@@ -24,6 +25,103 @@ pub const REDACTED: &str = "Redacted";
 /// Type alias for serde_json value which has Secret Information
 pub type SecretSerdeValue = Secret<serde_json::Value>;
 
+/// Phone Number
+#[derive(Debug, serde::Deserialize, serde::Serialize)]
+pub struct PhoneNumber(Secret<String>);
+
+impl<T> Strategy<T> for PhoneNumber
+where
+    T: AsRef<str> + std::fmt::Debug,
+{
+    fn fmt(val: &T, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let val_str: &str = val.as_ref();
+
+        if val_str.len() < 10 || val_str.len() > 12 {
+            return WithType::fmt(val, f);
+        }
+
+        write!(
+            f,
+            "{}{}{}",
+            &val_str[..2],
+            "*".repeat(val_str.len() - 5),
+            &val_str[(val_str.len() - 3)..]
+        )
+    }
+}
```

Comment:
Please use a different unit struct for `PhoneNumberStrategy`, implement `Strategy` for `PhoneNumberStrategy` and update `PhoneNumber` to use it instead, like so:

```rust
struct PhoneNumberStrategy;

impl Strategy for PhoneNumber {
    // ...
}

struct PhoneNumber(Secret<String, PhoneNumberStrategy>);
```

For more information, please refer to this comment: https://github.com/juspay/hyperswitch/pull/889#discussion_r1168201823